### ISSUE TITLE
Issues/Wiki are enabled only after first commit

### ISSUE THREAD

**Issue by steveeJ** [gogits/gogs#4104]
Gogs Version: 0.9.135.0208 running on https://try.gogs.io at the time of issue creation.

## Description
The Issues/Wiki featuers are enabled only after first commit. I think that this restriction is not helpful and I don't see a dependency between files in the repository and issues/wiki entries. I would even argue that the repository could be initialized by a first PR, but that is not my main argument here.
...

**Comment by Fastidious** [comment]
I am not understanding. So, you want to be able to open issues, but about what? `Issues` and `Wiki` are both about the contents of a repository. If the repository has no content (no commit, it is empty), then there are no issues, nor wiki to talk about.

This might be a feature/enhancement some might want, however.

**Comment by Unknwon** [comment]
Yes, empty repository doesn't make sense to have issues/PRs and wikis.

**Comment by steveeJ** [comment]
Code will not magically appear out of nowhere, but it will always start with an idea. Software like Gogs is used for collaborative work on software projects, which starts before the first line of code is written. 
A group of developers might just have an idea and wants to start organizing the project. The issue/milestone system can be used as a TODO list for the empty repository. The wiki system can be used for writing blueprints for the upcoming repository content.

@Unknwon 
I think that you closed this issue too quickly and unreasoned.

**Comment by Unknwon** [comment]
Good point.

**Comment by Unknwon** [comment]
Patch has pushed to fix this issue, please test on `develop` branch or https://try.gogs.io/unknwon/bare.

**Comment by steveeJ** [comment]
I tried with an empty repository I had created a while ago. I'm getting the following error on pushing the first commit:

```
remote: Gogs: Internal error
To try.gogs.io:steveeJ/alsoit.git
 ! [remote rejected] master -> master (pre-receive hook declined)
error: failed to push some refs to 'git@try.gogs.io:steveeJ/alsoit.git'
```

The commit looks as follows:

```diff
$ git show
commit 8d59b58dbbe160f48e30efec42fcbcaa58fed804
Author: Stefan Junker <mail@stefanjunker.de>
Date:   Fri Mar 24 14:03:29 2017 +0100

    Hi
    
    Fixes #1.

diff --git a/README.md b/README.md
new file mode 100644
index 0000000..76f177f
--- /dev/null
+++ b/README.md
@@ -0,0 +1 @@
+# Hi
```

**Comment by Unknwon** [comment]
@steveeJ since demo site is running all experimental/unstable versions... some old repos could just have corrupted data. :sweat_smile:
